src/engine/GroupBy.cpp

```diff
@@ -1077,36 +1102,65 @@ std::vector<size_t> GroupBy::HashMapAggregationData::getHashEntries(
 }
 
 // _____________________________________________________________________________
-[[nodiscard]] std::vector<Id>
-GroupBy::HashMapAggregationData::getSortedGroupColumn() const {
-  std::vector<ValueId> sortedKeys;
-  sortedKeys.reserve(map_.size());
-  // TODO<C++23>: use ranges::to
+template <size_t GROUPED_COLUMNS>
+[[nodiscard]] GroupBy::HashMapAggregationData<GROUPED_COLUMNS>::T<
+    std::vector<Id>>
+GroupBy::HashMapAggregationData<GROUPED_COLUMNS>::getSortedGroupColumns()
+    const {
+  // Get data in a row-wise manner.
+  std::vector<T<Id>> sortedKeys;
   for (const auto& val : map_) {
     sortedKeys.push_back(val.first);
   }
-  std::ranges::sort(sortedKeys);
-  return sortedKeys;
+
+  // Sort data.
+  std::ranges::sort(sortedKeys.begin(), sortedKeys.end());
+
+  // Get data in a column-wise manner.
+  T<std::vector<Id>> result;
+  if constexpr (requires { result.resize(numOfGroupedColumns_); })
+    result.resize(numOfGroupedColumns_);
+  for (size_t idx = 0; idx < numOfGroupedColumns_; ++idx)
+    for (auto& val : sortedKeys) {
+      result.at(idx).push_back(val.at(idx));
+    }
+
+  return result;
```

Comment:
What you should definitely do is at timers to your code:
1. One that only measures the (blockwise) hash-map lookup.
2. One that measures the Evaluation of the expressions.
3. One that measures the sorting of the final result.
4. One that measures the writing of the final result.

The result of each of these timers should at the end be added to the runtimeInfo

(`runtimeInfo().addDetail("timeResultSorting", resultTimer.msecs().count()`), 
same for the other measurements.
Then we can easily see in the UI, where the performance bottleneck lies, or in general where the time is spent.
(We have a timer class in `util/Timer` that is already widely used in QLever.